src/controls/qml/Particles.qml

```diff
@@ -0,0 +1,296 @@
+                var particle = component.createObject(meterFill, {
+                    "x": 10,
+                    "y": 5,
+                    "targetX": 50,
+                    "maxSize": 8,
+                    "lifetime": 1200,
+                    "isIncreasing": true,
+                    "design": "diamonds"
+                });
+            }
+        }
+
+        Timer {
+            interval: 200
+            running: true
+            repeat: true
+            onTriggered: createParticle()
+        }
+    }
+    \endqml
+*/
+Item {
+    id: particleRoot
+    width: maxSize
+    height: maxSize
+
+    /*!
+        \qmlproperty real Particles::maxSize
+        The maximum size of the particle.
+    */
+    property real maxSize: 10
+
+    /*!
+        \qmlproperty real Particles::targetX
+        The target X position for the particle's movement.
+    */
+    property real targetX: 0
+
+    /*!
+        \qmlproperty int Particles::lifetime
+        The duration (in milliseconds) before the particle self-destructs.
+    */
+    property int lifetime: 1200
+
+    /*!
+        \qmlproperty bool Particles::isIncreasing
+        Whether the value is increasing, affecting particle behavior.
+    */
+    property bool isIncreasing: false
+
+    /*!
+        \qmlproperty string Particles::design
+        The particle design: "diamonds", "bubbles", "logos", or "flashes".
+    */
+    property string design: "diamonds"
+
+    /*!
+        \qmlproperty rect Particles::clipBounds
+        The bounding rectangle (x, y, width, height) for clipping. Particles are destroyed if they move outside this area.
+    */
+    property rect clipBounds: Qt.rect(0, 0, 0, 0)
+
+    // Define design-specific properties
+    property var designProperties: {
+        "diamonds": { initialSize: 0.3, maxSize: 0.9, initialOpacity: 0, maxOpacity: 0.6 },
+        "bubbles": { initialSize: 0.3, maxSize: 0.9, initialOpacity: 0, maxOpacity: 0.6 },
+        "logos": { initialSize: 0.4, maxSize: 1.2, initialOpacity: 0, maxOpacity: 0.6 },
+        "flashes": { initialSize: 0.6, maxSize: 1.4, initialOpacity: 0, maxOpacity: 0.6 }
+    }
+
+    // Get design properties with fallback
+    function getDesignProp(propName) {
+        return (designProperties[design] && designProperties[design][propName])
+               ? designProperties[design][propName]
+               : designProperties["diamonds"][propName];
+    }
+
+    // Check if particle is outside clipBounds and destroy if so
+    onXChanged: {
+        if (clipBounds.width > 0 && clipBounds.height > 0) {
+            if (x < clipBounds.x - maxSize || x > clipBounds.x + clipBounds.width) {
+                particleRoot.destroy();
+            }
+        }
+    }
+
+    // Destroy timer to handle particle cleanup
+    Timer {
+        id: destroyTimer
+        interval: lifetime
+        running: true
+        repeat: false
+        onTriggered: particleRoot.destroy()
+    }
+
+    // Diamond design
+    Rectangle {
```

Comment:
I wonder if this can be optimized by storing these designs in a `Component`. Then you only need one way to show them using a `Loader`.

Advantages are:
-  Only one design is loaded at a time: more memory efficient
- Cleaner code, bu having designs separate from logic.

Disadvantages:
- Possible flicker when loading a new design. Might be mitigated with two loaders?
- Animations might be harder.

Reply:
Meh, too much work for little benefit, since only the 'design' is rendered the other should remain dormant.